src/main/java/org/programmers/springboot/basic/config/DataSourceConfig.java

```diff
@@ -0,0 +1,29 @@
+package org.programmers.springboot.basic.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.stereotype.Component;
+
+import javax.sql.DataSource;
+
+@Component
+public class DataSourceConfig {
+
+    private final DataSourceProperties dataSourceProperties;
+
+    @Autowired
+    public DataSourceConfig(DataSourceProperties dataSourceProperties) {
+        this.dataSourceProperties = dataSourceProperties;
+    }
+
+    @Bean
+    public DataSource mysqlDataSource() {
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
+        dataSource.setDriverClassName(dataSourceProperties.getDriverClassName());
+        dataSource.setUrl(dataSourceProperties.getUrl());
+        dataSource.setUsername(dataSourceProperties.getUsername());
+        dataSource.setPassword(dataSourceProperties.getPassword());
+        return dataSource;
+    }
```

Comment:
```suggestion
    @Bean
    public DataSource mysqlDataSource() {
        return DataSourceBuilder.create()
                .url(dataSourceProperties.getUrl())
                .driverClassName(dataSourceProperties.getDriverClassName())
                .username(dataSourceProperties.getUsername())
                .password(dataSourceProperties.getPassword())
                .build();
    }
```

Builder 를 더 많이 사용하는 것 같습니다.!

Reply:
DataSource에 대한 Builder가 있는지 알지 못했네요,, 다음에 사용하게 되면 Builder를 사용해보겠습니다. 감사합니다!